### PR TITLE
Improve logging  - Move to Slog fix #332

### LINKED ISSUE #332: Improve logging
The project isn't consistent when it comes to logging. We should choose a logging library that will give users a standard logs which will allow them integrate the logs into their preferred logging aggregator and filter them to their needs.

Some popular options:
1. logrus
2. zap
3. zerolog

opinions are welcome in this discussion :)

### PR DESCRIPTION
**Pull request by avivl**
fix #332

### CODE REVIEW

**Comment by avivl** [comment]
@smintz Tests are also failing on master branch
```
 module_service_test.go:149: ModuleService.Sync() error = failed to validate integrity
        com_github_protoconf_protoconf_xds(&{27d699b}): h1:MHam+LpdxRMBNGdU+WzfqO6HWpMNfS/6ZHEU+nI1KpI= (expected: h1:0pcqbr1XMmwU2HAweYk9o3Z/6EGZ69Y8yR9GGANEXqs=), wantErr <nil>
 ```

**Comment by avivl** [comment]
> @smintz Tests are also failing on master branch
> 
> ```
>  module_service_test.go:149: ModuleService.Sync() error = failed to validate integrity
>         com_github_protoconf_protoconf_xds(&{27d699b}): h1:MHam+LpdxRMBNGdU+WzfqO6HWpMNfS/6ZHEU+nI1KpI= (expected: h1:0pcqbr1XMmwU2HAweYk9o3Z/6EGZ69Y8yR9GGANEXqs=), wantErr <nil>
> ```

Fixed

File: compiler/template.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func evaluateTemplateFile(filename string) error {
 
 	targetFile := filepath.Join(consts.SrcPath, strings.TrimSuffix(filename, `.template`))
 
-	log.Println("writing to", targetFile)
+	slog.Info("writing to", "targetfile", targetFile)
```
Comment (smintz) [inline compiler/template.go:91]: This should probably reduced to debug

File: compiler/template.go [hunk 2]
```diff
@@ -97,13 +97,13 @@ func evaluateTemplateFile(filename string) error {
 }
 
 func findTemplateFilesAndProccess() error {
-	log.Println("processing template")
+	slog.Info("processing template")
```
Comment (smintz) [inline compiler/template.go:100]: need to be reduced to debug

**Pull request merged**